### PR TITLE
ci: switch to GitHub Actions

### PR DESCRIPTION
**Pull request by kyleam**
This series switches the CI from Drone to GitHub Actions.

 * The main switch happens in the third commit.

 * The first commit tightens the `install_torsten()` skips to hopefully avoid gh-125.  This isn't directly related, but I'm trying to avoid spurious failures as iterating/troubleshooting this switch to GitHub Actions.

---

- [x] Drop tip commit that tests dry-run s3 sync.

### CODE REVIEW

**Comment by kyleam** [comment]
Test s3 auth and dry-run upload look good:

 * https://github.com/metrumresearchgroup/bbr.bayes/actions/runs/8901716835/job/24446525873?pr=142
 * b488248878b70cb346b85b1cf41a96c831cdc271

I'll drop the tip commit.

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+      - 'main'
+      - 'scratch/**'
+    tags:
+      - '[0-9]+.[0-9]+.[0-9]+'
+      - '[0-9]+.[0-9]+.[0-9]+.[0-9]+'
+  pull_request:
+
+env:
+  BBI_VERSION: v3.3.0
+  MPN: 2024-03-01
```
Comment (kyleam) [inline .github/workflows/main.yaml:14]: This hard coding of the latest MPN version is unfortunate.  Drone, instead, just uses the `:latest` and `:cran-latest` Docker tags, and those get updated behind the scenes.  Each package doesn't need to handle the update.

In the near future (ideally the next MPN release), I think as part of publishing the MPN site, we should update something like `https://mpn.metworx.com/latest` to provide an easy way to programmatically get the latest MPN version.  (There is a way we've used internally for one script, but it's really convoluted, and I'd rather not propagate it to the GitHub Actions config for each of our R packages.)

Comment (kyleam) [reply .github/workflows/main.yaml]: Update to query new `https://mpn.metworx.com` path.

**Review by seth127 — APPROVED** [review]
Good stuff. Thanks.

**Pull request merged**